crowbar_framework/app/models/node.rb

```diff
@@ -24,8 +24,12 @@ class Node < ChefObject
 
   self.chef_type = "node"
 
-  def initialize(node)
-    @role = RoleObject.find_role_by_name Node.make_role_name(node.name)
+  def initialize(node, role = nil)
+    @role = if role.nil?
+      RoleObject.find_role_by_name Node.make_role_name(node.name)
+    else
+      role
```

Comment:
Maybe some paranoid check that the role name matches `Node.make_role_name(node.name)` would be good?

Reply:
This would be good if we have some kind of assertions as this would most likely be a coding error. I'd skip that for now.